### ISSUE TITLE
Train RTM-pose on custom dataset, got strange loss in log, and strange inference result 

### ISSUE THREAD

**Issue by lu-minous** [open-mmlab/mmpose#2189]
Thanks for your error report and we appreciate it a lot.
If you feel we have helped you, give us a STAR! :satisfied:

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. The bug has not been fixed in the latest version.

**Describe the bug**

Train RTM-pose on custom dataset, got strange loss in log, and strange inference result.

1. loss 

```bash
python tools/train.py projects/rtmpose/rtmpose/body_2d_keypoint/rtmpose-s_1xb128-240e_CCPDminicoco-256x192.py --show-dir visual/eval/ 
```

- 1-36 epochs: The loss is reduced to about 1.5 and no longer changes? (lr: 4e-3, 1e-3, 4e-4, 1e-4)
- After 38 epochs: The loss gradually decreases to 0.23, but eval PCK(thr=0.1), AUC have not changed much(or rising slightly)?
- Last 30 epoch: PCK(thr=0.1) reached 0.6, AUC reached 0.7.(Data Aug is softer)
2. infer
```
python demo/image_demo.py \
    /content/bboxl1.jpg \
    projects/rtmpose/rtmpose/body_2d_keypoint/rtmpose-s_1xb128-240e_CCPDminicoco-256x192.py \
    work_dirs/rtmpose-s_1xb128-240e_CCPDminicoco-256x192/epoch_160.pth \
    --out-file ccpd_vis_results_256192_3.jpg \
    --draw-heatmap
```
![image](https://user-images.githubusercontent.com/48793299/230677259-6be342b5-0cd2-4090-9fc2-7bf1b4b12cd9.png)

WHAT I DO:

1. Prepare my dataset in coco format

![image](https://user-images.githubusercontent.com/48793299/230666866-e83be8fc-1c60-4da0-9daf-6eb02f6ebd20.png)

image id: 6316
![0312662760417-91_95-168 477_507 570-507 570_181 562_168 477_502 480-0_0_3_24_32_31_31_33-104-203](https://user-images.githubusercontent.com/48793299/230666326-a80fb9a4-db52-4864-a280-d8606399ae44.jpg)
train json:
![image](https://user-images.githubusercontent.com/48793299/230667241-64f5b936-d9b6-46d6-af40-57f578ce4cb6.png)

2. Add /content/mmpose/mmpose/datasets/datasets/body/ccpdcoco_dataset.py:

```python
# Copyright (c) OpenMMLab. All rights reserved.
from mmpose.registry import DATASETS
from ..base import BaseCocoStyleDataset


@DATASETS.register_module()
class CCPDCocoDataset(BaseCocoStyleDataset):


    METAINFO: dict = dict(from_file='configs/_base_/datasets/ccpdcoco.py')
    
```
3. Add 'CCPDCocoDataset' /content/mmpose/mmpose/datasets/datasets/body/init.py
4. Copy rtmpose-s config and modify it: 

**Reproduction**

My code:
log:
[230408.log](https://github.com/open-mmlab/mmpose/files/11183844/230408.log)
colab notebook:
https://colab.research.google.com/drive/1UJES_s6CR-SFWvfNoTb6cZSh9kznqqnK?usp=sharing
mmpose dir:
https://drive.google.com/drive/folders/12k3VUARz57ZDj6ya4nrD3YrVSvgjlMuk?usp=sharing
dataset:

- What config did you run?

```python
_base_ = ['mmpose::_base_/default_runtime.py']

# runtime
max_epochs = 180
stage2_num_epochs = 30
base_lr = 1e-4 # 4e-3

train_cfg = dict(max_epochs=max_epochs, val_interval=10)
randomness = dict(seed=21)

# optimizer
optim_wrapper = dict(
    type='OptimWrapper',
    optimizer=dict(type='AdamW', lr=base_lr, weight_decay=0.),
    paramwise_cfg=dict(
        norm_decay_mult=0, bias_decay_mult=0, bypass_duplicate=True))

# learning rate
param_scheduler = [
    dict(
        type='LinearLR',
        start_factor=1.0e-5,
        by_epoch=False,
        begin=0,
        end=2000),
    dict(
        # use cosine lr from 210 to 420 epoch
        type='CosineAnnealingLR',
        eta_min=base_lr * 0.05,
        begin=max_epochs // 2,
        end=max_epochs,
        T_max=max_epochs // 2,
        by_epoch=True,
        convert_to_iter_based=True),
]

# automatically scaling LR based on the actual training batch size
auto_scale_lr = dict(base_batch_size=1024)

# codec settings
codec = dict(
    type='SimCCLabel',
    #input_size=(192, 256),
    input_size=(256, 192),
    #sigma=(4.9, 5.66),
    sigma=(5.66, 4.9),
    simcc_split_ratio=2.0,
    normalize=False,
    use_dark=False)

# model settings
model = dict(
    type='TopdownPoseEstimator',
    data_preprocessor=dict(
        type='PoseDataPreprocessor',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        bgr_to_rgb=True),
    backbone=dict(
        _scope_='mmdet',
        type='CSPNeXt',
        arch='P5',
        expand_ratio=0.5,
        deepen_factor=0.33,
        widen_factor=0.5,
        out_indices=(4, ),
        channel_attention=True,
        norm_cfg=dict(type='SyncBN'),
        act_cfg=dict(type='SiLU'),
        
        #init_cfg=dict(
        #    type='Pretrained',
        #    prefix='backbone.',
        #    checkpoint='https://download.openmmlab.com/mmpose/v1/projects/'
        #    'rtmpose/cspnext-s_udp-aic-coco_210e-256x192-92f5a029_20230130.pth'  # noqa
        #    )
        
        ),
    head=dict(
        type='RTMCCHead',
        in_channels=512,
        #out_channels=17,
        out_channels=4,
        input_size=codec['input_size'],
        in_featuremap_size=(8, 6),
        simcc_split_ratio=codec['simcc_split_ratio'],
        final_layer_kernel_size=7,
        gau_cfg=dict(
            hidden_dims=256,
            s=128,
            expansion_factor=2,
            dropout_rate=0.,
            drop_path=0.,
            act_fn='SiLU',
            use_rel_bias=False,
            pos_enc=False),
        loss=dict(
            type='KLDiscretLoss',
            use_target_weight=True,
            beta=10.,
            label_softmax=True),
        decoder=codec),
    #test_cfg=dict(flip_test=True))
    test_cfg=dict(flip_test=False))

# base dataset settings
dataset_type = 'CCPDCocoDataset'
data_mode = 'topdown'
#data_root = 'data/coco/'data/CCPD2020/ccpd_green'
data_root = '/content/CCPDpose/'

file_client_args = dict(backend='disk')
# file_client_args = dict(
#     backend='petrel',
#     path_mapping=dict({
#         f'{data_root}': 's3://openmmlab/datasets/detection/coco/',
#         f'{data_root}': 's3://openmmlab/datasets/detection/coco/'
#     }))

# pipelines
train_pipeline = [
    dict(type='LoadImage', file_client_args=file_client_args),
    dict(type='GetBBoxCenterScale'),
    #dict(type='RandomFlip', direction='horizontal'),
    dict(type='RandomHalfBody'),
    dict(
        #ype='RandomBBoxTransform', scale_factor=[0.6, 1.4], rotate_factor=80), # bbox resize rotate(°)
        type='RandomBBoxTransform', scale_factor=[0.7, 1.3], rotate_factor=20),
    dict(type='TopdownAffine', input_size=codec['input_size']),
    dict(type='mmdet.YOLOXHSVRandomAug'),
    dict(
        type='Albumentation',
        transforms=[
            dict(type='Blur', p=0.1),
            dict(type='MedianBlur', p=0.1),
            dict(
                type='CoarseDropout',
                max_holes=1,
                max_height=0.4,
                max_width=0.4,
                min_holes=1,
                min_height=0.2,
                min_width=0.2,
                p=1.),
        ]),
    dict(type='GenerateTarget', encoder=codec),
    dict(type='PackPoseInputs')
]
val_pipeline = [
    dict(type='LoadImage', file_client_args=file_client_args),
    dict(type='GetBBoxCenterScale'),
    dict(type='TopdownAffine', input_size=codec['input_size']),
    dict(type='PackPoseInputs')
]

train_pipeline_stage2 = [
    dict(type='LoadImage', file_client_args=file_client_args),
    dict(type='GetBBoxCenterScale'),
    #dict(type='RandomFlip', direction='horizontal'),
    dict(type='RandomHalfBody'),
    dict(
        type='RandomBBoxTransform',
        shift_factor=0.,
        #cale_factor=[0.75, 1.25], #bbox resize
        scale_factor=[0.8, 1.2],
        #rotate_factor=60
        rotate_factor=10
        ),
    dict(type='TopdownAffine', input_size=codec['input_size']),
    dict(type='mmdet.YOLOXHSVRandomAug'),
    dict(
        type='Albumentation',
        transforms=[
            dict(type='Blur', p=0.1),
            dict(type='MedianBlur', p=0.1),
            dict(
                type='CoarseDropout',
                max_holes=1,
                max_height=0.4,
                max_width=0.4,
                min_holes=1,
                min_height=0.2,
                min_width=0.2,
                p=0.5),
        ]),
    dict(type='GenerateTarget', encoder=codec),
    dict(type='PackPoseInputs')
]

# data loaders
train_dataloader = dict(
    batch_size=128, #256
    num_workers=2,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=True),
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        data_mode=data_mode,
        ann_file='anno/ccpdmini_annotations_train.json',
        data_prefix=dict(img='train/'),
        pipeline=train_pipeline,
    ))
val_dataloader = dict(
    batch_size=128,
    num_workers=2,
    persistent_workers=True,
    drop_last=False,
    sampler=dict(type='DefaultSampler', shuffle=False, round_up=False),
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        data_mode=data_mode,
        ann_file='anno/ccpdmini_annotations_val.json',
        # bbox_file=f'{data_root}person_detection_results/'
        # 'COCO_val2017_detections_AP_H_56_person.json',
        data_prefix=dict(img='val/'),
        test_mode=True,
        pipeline=val_pipeline,
    ))
test_dataloader = val_dataloader

# hooks
default_hooks = dict(
    #checkpoint=dict(save_best='coco/AP', rule='greater', max_keep_ckpts=1))
    checkpoint=dict(save_best='AUC', rule='greater', max_keep_ckpts=1))

custom_hooks = [
    dict(
        type='EMAHook',
        ema_type='ExpMomentumEMA',
        momentum=0.0002,
        update_buffers=True,
        priority=49),
    dict(
        type='mmdet.PipelineSwitchHook',
        switch_epoch=max_epochs - stage2_num_epochs,
        switch_pipeline=train_pipeline_stage2)
]

# evaluators
val_evaluator = [
    dict(type='PCKAccuracy', thr=0.1),
    dict(type='AUC'),
    dict(type='EPE')
]
#val_evaluator = dict(
#    type='CocoMetric',
#    ann_file=data_root + '../code/ccpdmini_annotations_test.json')
test_evaluator = val_evaluator

```

**Environment**

1. Please run `PYTHONPATH=${PWD}:$PYTHONPATH python mmpose/utils/collect_env.py` to collect necessary environment information and paste it here.
```
------------------------------------------------------------
System environment:
    sys.platform: linux
    Python: 3.9.16 (main, Dec  7 2022, 01:11:51) [GCC 9.4.0]
    CUDA available: True
    numpy_random_seed: 21
    GPU 0: Tesla T4
    CUDA_HOME: /usr/local/cuda
    NVCC: Cuda compilation tools, release 11.8, V11.8.89
    GCC: x86_64-linux-gnu-gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
    PyTorch: 2.0.0+cu118
    PyTorch compiling details: PyTorch built with:
  - GCC 9.3
  - C++ Version: 201703
  - Intel(R) oneAPI Math Kernel Library Version 2022.2-Product Build 20220804 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.7.3 (Git Hash 6dbeffbae1f23cbbeae17adb7b5b13f1f37c080e)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.8
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_90,code=sm_90
  - CuDNN 8.7
  - Magma 2.6.1
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.8, CUDNN_VERSION=8.7.0, CXX_COMPILER=/opt/rh/devtoolset-9/root/usr/bin/c++, CXX_FLAGS= -D_GLIBCXX_USE_CXX11_ABI=0 -fabi-version=11 -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOROCTRACER -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -O2 -fPIC -Wall -Wextra -Werror=return-type -Werror=non-virtual-dtor -Werror=bool-operation -Wnarrowing -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wunused-local-typedefs -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_DISABLE_GPU_ASSERTS=ON, TORCH_VERSION=2.0.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=1, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, 

    TorchVision: 0.15.1+cu118
    OpenCV: 4.7.0
    MMEngine: 0.7.2

Runtime environment:
    cudnn_benchmark: False
    mp_cfg: {'mp_start_method': 'fork', 'opencv_num_threads': 0}
    dist_cfg: {'backend': 'nccl'}
    seed: 21
    Distributed launcher: none
    Distributed training: False
    GPU number: 1
------------------------------------------------------------
```

4. You may add addition that may be helpful for locating the problem, such as

- How you installed PyTorch \[e.g., pip, conda, source\]
- Other environment variables that may be related (such as `$PATH`, `$LD_LIBRARY_PATH`, `$PYTHONPATH`, etc.)

**Error traceback**

If applicable, paste the error traceback here.

```
A placeholder for traceback.
```

**Bug fix**

If you have already identified the reason, you can provide the information here. If you are willing to create a PR to fix it, please also leave a comment here and that would be much appreciated!

**Comment by lu-minous** [comment]
It seems that my handmade CCPDminipose(crop+resize+keypoint) dataset is questionable?
1. Some images in json are redundant, because I delete some invalid(keypoint is beyond image) annos.
2. Every anno has same id 1, does it matter ?(As each image in CCPD contains only a single license plate (LP).)

**Comment by lu-minous** [comment]
There is new progress!!
I switched to a new dataset and got very correct result(both in loss and inference).
